<launch>

  <!-- Load hand geometry parameters -->
  <include file="$(find gpd)/launch/hand_geometry.launch">
    <arg name="node" value="classify_grasp_candidates" />
  </include>

  <!-- Load classifier parameters -->
  <include file="$(find gpd)/launch/lenet/ur5_classifier_15channels.launch">
    <arg name="node" value="classify_grasp_candidates" />
  </include>

	<node name="classify_grasp_candidates" pkg="gpd" type="classify_candidates" output="screen">

    <!-- Where the point cloud is stored -->
    <param name="cloud_file_name" value="$(find gpd)/tutorials/krylon.pcd" />

    <!-- <param name="cloud_file_name" value="/home/andreas/data/handle_detector/Baxter/table10.pcd" /> -->


    <!-- Plotting parameters -->
    <param name="plot_normals" value="true" />
    <param name="plot_samples" value="false" />
    <param name="plot_candidates" value="true" />
    <param name="plot_filtered_grasps" value="false" />
    <param name="plot_valid_grasps" value="true" />
    <param name="plot_clusters" value="false" />
    <param name="plot_selected_grasps" value="false" />

    <!-- Preprocessing of point cloud -->
    <param name="voxelize" value="false"/> <!-- Voxelize the point cloud -->
    <param name="remove_outliers" value="false"/> <!-- Remove statistical outliers from the point cloud -->
    <!-- <rosparam param="workspace"> [-1, 1, -1, 1, 0, 1] </rosparam> -->
    <rosparam param="workspace"> [-1, 1, -1, 1, -1, 1] </rosparam> -->
    <rosparam param="camera_position"> [0, 0, 0] </rosparam> <!-- The position of the camera -->

    <!-- General parameters -->
    <param name="num_samples" value="100" /> <!-- The number of samples to draw from the point cloud -->
    <param name="num_threads" value="4" /> <!-- The number of CPU threads to be used -->

    <!-- Parameters for local grasp candidate search -->
    <param name="nn_radius" value="0.01" /> <!-- Radius for the point neighborhood search -->
    <param name="num_orientations" value="8" /> <!-- Number of orientations to consider -->

    <!-- Filtering of grasp candidates -->
    <param name="filter_grasps" value="false" /> <!-- on workspace and robot gripper width -->
    <param name="min_aperture" value="0.3" /> <!-- The minimum gripper width -->
    <param name="max_aperture" value="0.7" /> <!-- The maximum gripper width -->
    <rosparam param="workspace_grasps"> [-1, 1, -1, 1, -1, 1] </rosparam> <!-- (x,y,z)-dimensions of the workspace, should be smaller than <workspace> -->
    <param name="filter_half_antipodal" value="false"/> <!-- on half antipodal -->
    
    <!-- Filtering of grasps which are too low on (i.e. too close to) table -->
    <param name="filter_table_side_grasps" value="false" />
    <rosparam param="vertical_axis"> [0, 0, 1] </rosparam>
    <param name="angle_thresh" value="0.1" />
    <param name="table_height" value="0.0" />
    <param name="table_thresh" value="0.05" />

    <!-- Grasp image creation -->
    <param name="create_image_batches" value="false" /> <!-- creates grasp images in batches (less memory usage) -->

    <!-- Clustering of grasps -->
    <param name="min_inliers" value="10" /> <!-- min inliers in each cluster; set to 0 to avoid clustering -->

    <!-- Grasp selection -->
    <param name="min_score_diff" value="0" /> <!-- Threshold to sort out grasps with low scores -->
    <param name="num_selected" value="100" /> <!-- The number of selected grasps (sorted by score) -->

	</node>

</launch>
